File: Helmut/6/solve.py
t = open("input","r").read()
# t = "zcfzfwzzqfrljwzlrfnpqdbhtmscgvjw"
for i in range(len(t)-14):
    if len(set(t[i:i+14]))==14:
        print(i+14)
        break
        

File: Helmut/6/solve.hs
import Data.List (tails, nub)

check :: String -> Int
check = (4+) . length . takeWhile (<4) . map (length . nub . take 4) . tails

main = do
    print $ check "zcfzfwzzqfrljwzlrfnpqdbhtmscgvjw"
    print $ check "nppdvjthqldpwncqszvftbrmjlhg"
    fmap check (readFile "input") 
